File: leetcode/easy/20_Valid_Parentheses.ts
function isValid(s: string): boolean {
  const myhash: { [key: string]: string } = { "(": ")", "{": "}", "[": "]" };
  const mystack: string[] = [];
  for (const char of s) {
    if (char in myhash) {
      mystack.push(char);
    } else {
      if (mystack.length === 0) {
        return false;
      }

      const tmp = mystack.pop() as string;
      if (char !== myhash[tmp]) {
        return false;
      }
    }
  }
  if (mystack.length !== 0) {
    return false;
  }
  return true;
}

File: leetcode/easy/20_Valid_Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = [s[0]]
        pair_brackets = {
            ']': '[',
            ')':'(',
            '}':'{',
        }
        i = 1
        while i < len(s):
            if s[i] in pair_brackets.values():
                stack.append(s[i])
            elif s[i] in pair_brackets.keys() and len(stack) > 0:
                checked = stack.pop()
                if pair_brackets[s[i]] == checked:
                    i+=1
                    continue
                else:
                    return False
            else:
                return False
            i += 1
        if len(stack) > 0:
            return False
        return True
